### ISSUE TITLE
react-bootstrap: using React.HTMLProps instead of React.Props

### ISSUE THREAD

**Issue by ianks** [DefinitelyTyped/DefinitelyTyped#8509]
- [x] I tried using the latest `react-bootstrap/react-bootstrap.d.ts` file in this repo and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] I want to talk about `react-bootstrap/react-bootstrap.d.ts`.
  - The authors of that type definition are @walkerburgin, @vsiao 
## Issue

In the typings for react-bootstrap, I noticed that the typings were extending the interface for `React.Props` in most cases. 

i.e.

``` typescript
interface PanelProps extends React.Props<Panel> {
    className?: string;
    bsSize?: string;
    bsStyle?: string;
    collapsible?: boolean;
    defaultExpanded?: boolean;
    eventKey?: any;
    expanded?: boolean;
    footer?: any; // TODO: Add more specific type
    header?: any; // TODO: Add more specific type
    id?: string;
    onSelect?: Function; // TODO: Add more specific type
    onClick?: Function; // TODO: Add more specific type
}
```

This is is problematical for a few reasons:
1.  Valid code is incorrectly type-checked:

``` typescript
<Panel style={{ borderLeftColor: "#FFF" }}/>

// error TS2339: Property 'style' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<PanelProps, any>> & PanelProps'.
```
1. We lose specificity/compatibility with React HTMLProps

For example, instead of having:

``` typescript
interface PanelProps extends React.Props<Panel> {
    // ...
    onSelect?: Function; // TODO: Add more specific type
   // ...
}
```

If we instead extend `React.HTMLProps<Panel>`, our `onSelect` type signature becomes: `onSelect?: ReactEventHandler;`, with zero work on our part.
## tl;dr

Should we extend `React.HTMLProps<T>` instead on `React.Props<T>`? What are the pros and cons? I am happy to take on this endeavor if it is decided that it would be beneficial.

**Comment by vsiao** [comment]
You'll want to extend `React.HTMLAttributes` (`React.Props<T>` has been deprecated as of #8543).
